ui/src/components/StdTable.tsx

```diff
@@ -0,0 +1,205 @@
+    const endIndex = startIndex + itemsPerPage;
+    const paginatedData = filteredData.slice(startIndex, endIndex);
+    
+    const handlePageChange = (selectedPage: { selected: number }) => {
+      setCurrentPage(selectedPage.selected);
+    };
+     
+    const totalPages = Math.ceil(filteredData.length / itemsPerPage);
+
+  return (
+    <div className="mt-1">
+      
+      {Editbool  || AddModal ? (
+        <Modal body={
+          <>
+          {AddModal ? <StdTableAddEdit handle={()=>{setAddModal(false)}}/> : null}
+          {Editbool ? <StdTableAddEdit handle ={()=>seteditbool(false)} initialData={EditModal}/> : null} 
+        </>
+      }
+      header = {
+        {title : AddModal ? 'Add data' : 'Edit data',
+        closehandle : AddModal ? ()=>{setAddModal(false)} : ()=>{seteditbool(false)}}
+      }
+      /> 
+      ) : null}
+      
+       <div className="flex justify-evenly items-center mb-4">
+        <label htmlFor="search" className="block ml-6 text-sm font-medium text-gray-700">
+          Search 
+        </label>
+        <input
+          type="text"
+          id="search"
+          placeholder="Search..."
+          value={searchQuery}
+          onChange={(e) => search(e)}
+          className="mt-1 p-2  ml-5 w-96 border rounded-md focus:ring focus:ring-blue-200"
+        />
+        
+          <button className="bg-Aqua-300 text-Black font-medium m-1 py-2 px-4 rounded-md hover:font-bold ml-2"
+            onClick={() => setAddModal(true)}>
+            ADD
+          </button>
+          
+         
+      </div>
+      {filteredData.length != 0 ? (
+        <>
+        
+        <table className="w-11/12 mx-auto border-rounded border border-gray-300 shadow-md">
+        <thead>
+          <tr className="bg-Aqua-300 text-Black shadow-md ">
+            <th className="p-3 text-center">Student ID</th>
+            <th className="p-3 text-center">Name</th>
+            <th className="p-3 text-center">Roll no.</th>
+            <th className="p-3 text-center">Class</th>
+            <th className="p-3 text-center">Actions</th>
+          </tr>
+        </thead>
+        <tbody>
+          {paginatedData.map((data, index) => (
+            <tr key={index} className={index % 2 === 0 ? 'bg-gray-100' : ''}>
+              <td className="p-3 text-center">{data.username}</td>
+              <td className="p-3 text-center">{data.name}</td>
+              <td className="p-3 text-center">{data.rollnumber}</td>
+              <td className="p-3 text-center">{data.classname}</td>
+              <td className="p-3 text-center">
+              <div className='grid grid-cols-2 gap-3'>
+              <MdEdit onClick={()=>
+                {setEditModal(data)
+                 seteditbool(true)
+                }}/>
+              <MdDelete onClick={() => OnDelete(data.id)}/>
+              </div>          
+              </td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
```

Comment:
We can have a Table component which will take data and display. So that We don't need to write Table everywhere

Reply:
Added